File: LeetcodeJava/src/dynamicprogramming/ClimbStairs.java
package dynamicprogramming;

public class ClimbStairs {
    public int climbStairs(int n) {
        int[] d = new int[n + 1];
        d[1] = 1;
        if (n >= 2)
            d[2] = 2;
        for (int i = 3; i <= n; i++) {
            d[i] = d[i - 1] + d[i - 2];
        }

        return d[n];
    }
}

File: LeetcodeKt/src/dynamic/ClimbStairs.kt
package dynamic

/*
#70.爬楼梯
 */
class ClimbStairs {
    fun climbStairs(n: Int): Int {
        if (n < 3) return n
        var p = 1
        var q = 2
        var r = 3
        for (i in 3 until n) {
            p = q
            q = r
            r = p + q
        }
        return r
    }
}

fun main() {
    ClimbStairs().climbStairs(45).let(::println)
}